cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -99,60 +77,55 @@ class GcsTestbench : public ::testing::Environment {
         "Could not start GCS emulator."
         " Used the following list of python interpreter names:");
     for (const auto& interpreter : names) {
-      auto exe_path = bp::search_path(interpreter);
+      auto server_process = std::make_unique<util::Process>();
       error += " " + interpreter;
-      if (exe_path.empty()) {
+      auto status = server_process->SetExecutable(interpreter);
+      if (!status.ok()) {
         error += " (exe not found)";
         continue;
       }
 
-      bp::ipstream output;
-      server_process_ = bp::child(exe_path, "-m", "testbench", "--port", port_, group_,
-                                  bp::std_err > output);
-      // Wait for message: "* Restarting with"
-      auto testbench_is_running = [&output, this](bp::child& process) {
-        std::string line;
-        std::chrono::time_point<std::chrono::steady_clock> end =
-            std::chrono::steady_clock::now() + std::chrono::seconds(10);
-        while (server_process_.valid() && server_process_.running() &&
-               std::chrono::steady_clock::now() < end) {
-          if (output.peek() && std::getline(output, line)) {
-            std::cerr << line << std::endl;
-            if (line.find("* Restarting with") != std::string::npos) return true;
-          } else {
-            std::this_thread::sleep_for(std::chrono::milliseconds(20));
-          }
-        }
-        return false;
-      };
-
-      if (testbench_is_running(server_process_)) break;
-      error += " (failed to start)";
-      server_process_.terminate();
-      server_process_.wait();
+      status = server_process->SetArgs({"-m", "testbench", "--port", port_});
+      if (!status.ok()) {
```

Comment:
Is it useful to add this information, especially as `SetArgs` cannot actually fail? (same question below)